Tribler/community/market/community.py

```diff
@@ -0,0 +1,1220 @@
+from Tribler.community.market.core.timestamp import Timestamp
+from Tribler.community.market.core.trade import Trade, ProposedTrade, DeclinedTrade, CounterTrade
+from Tribler.community.market.core.transaction import StartTransaction, TransactionId, Transaction
+from Tribler.community.market.core.transaction_manager import TransactionManager
+from Tribler.community.market.core.transaction_repository import DatabaseTransactionRepository, MemoryTransactionRepository
+from Tribler.community.market.core.ttl import Ttl
+from Tribler.community.market.core.wallet_address import WalletAddress
+from Tribler.community.market.database import MarketDB
+from Tribler.community.market.payload import OfferPayload, TradePayload, DeclinedTradePayload,\
+    StartTransactionPayload, TransactionPayload, WalletInfoPayload, MarketIntroPayload, OfferSyncPayload,\
+    PaymentPayload, CancelOrderPayload
+from Tribler.community.market.reputation.pagerank_manager import PagerankReputationManager
+from Tribler.community.market.wallet.tc_wallet import TrustchainWallet
+from Tribler.dispersy.authentication import MemberAuthentication
+from Tribler.dispersy.bloomfilter import BloomFilter
+from Tribler.dispersy.candidate import Candidate, WalkCandidate
+from Tribler.dispersy.community import Community
+from Tribler.dispersy.conversion import DefaultConversion
+from Tribler.dispersy.destination import CommunityDestination, CandidateDestination
+from Tribler.dispersy.distribution import DirectDistribution
+from Tribler.dispersy.message import Message, DelayMessageByProof, DropMessage
+from Tribler.dispersy.requestcache import IntroductionRequestCache, NumberCache
+from Tribler.dispersy.resolution import PublicResolution
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.internet.task import LoopingCall, deferLater
+
+
+class ProposedTradeRequestCache(NumberCache):
+    """
+    This cache keeps track of outstanding proposed trade messages.
+    """
+    def __init__(self, community, proposed_trade):
+        super(ProposedTradeRequestCache, self).__init__(community.request_cache, u"proposed-trade",
+                                                        proposed_trade.proposal_id)
+        self.community = community
+        self.proposed_trade = proposed_trade
+
+    def on_timeout(self):
+        # Just remove the reserved quantity from the order
+        order = self.community.order_manager.order_repository.find_by_id(self.proposed_trade.order_id)
+        order.release_quantity_for_tick(self.proposed_trade.recipient_order_id, self.proposed_trade.quantity)
+        self.community.order_manager.order_repository.update(order)
+
+        self._logger.debug("Proposed trade timed out, trying to find a new match for this order")
+        self.community.order_book.remove_tick(self.proposed_trade.recipient_order_id)
+        self.community.match(order)
+
+
+class MarketCommunity(Community):
```

Comment:
Could you add a `get_wallet_ids()` function to get the known wallet ids? Right now "users" need to approach wallets by accessing the wallet `variable` directly (maybe also add a `get_wallet(id)`).